src/shims/posix/fs.rs

```diff
@@ -827,7 +847,15 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
     ) -> InterpResult<'tcx, i32> {
         let this = self.eval_context_mut();
         this.assert_target_os("macos", "stat");
-        this.check_no_isolation("`stat`")?;
+
+        // Reject if isolation is enabled.
+        if let IsolatedOp::Reject(reject_with) = this.machine.isolated_op {
+            this.reject_in_isolation("`stat`", reject_with)?;
+            // macos stat never sets "EPERM". Set error code "ENOENT".
+            this.set_last_error_from_io_error(ErrorKind::NotFound)?;
```

Comment:
Is there no more accurate error code? What do macOS sandboxes typically do?

Reply:
This variant of `stat` works on a path name and doesn't need read, write or execute permission. `ENOENT` corresponds to file does not exist. Another relevant code is `EACCES` -- permission denied for a component of the path prefix. I'm not sure about macOS sandboxes. I guess `EACCES` sounds closer to `EPERM`.

Reply:
yeah EACCESS would be better I think